pkg/asset/installconfig/aws/subnet.go

```diff
@@ -0,0 +1,165 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var vpc string
+var vpcFromSubnet string
+var routeTables []*ec2.RouteTable
+var subnetMetadata map[string]*Subnet
+
+// Subnet hold metadata for a subnet.
+type Subnet struct {
+	// ARN is the subnet's Amazon Resource Name.
+	ARN string
+
+	// Public is true if the cloud-provider logic considers the subnet
+	// to be public (e.g. with suitable routing for hosting public
+	// load balancers).
+	Public bool
+
+	// VPC is the ID of the VPC to which the subnet belongs.
+	VPC string
+
+	// Zone is the subnet's availability zone.
+	Zone string
+}
+
+// SubnetMetadata retrieves metadata for the given subnet(s).  Results
+// are cached, so additional requests for a given subnet do not result
+// in further AWS API calls.
+func SubnetMetadata(ctx context.Context, ids []string) (map[string]Subnet, error) {
+	remainingIDs := make([]*string, 0, len(ids))
+	metas := make(map[string]Subnet, len(ids))
+	for _, id := range ids {
+		if meta, ok := subnetMetadata[id]; ok {
+			metas[id] = *meta
+		} else {
+			remainingIDs = append(remainingIDs, aws.String(id))
+		}
+	}
+	if len(remainingIDs) == 0 {
+		return metas, nil
+	}
+
+	session, err := GetSession()
+	if err != nil {
+		return metas, err
+	}
+
+	client := ec2.New(session)
+
+	var loopError error
+	results, err := client.DescribeSubnetsWithContext( // FIXME: port to DescribeSubnetsPagesWithContext once we bump our vendored AWS package past v1.19.30
```

Comment:
https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeSubnetsOutput is paged. so we might be missing some results.

Also bumping to atleast v1.19.30 should be fine.. any reason why not?

Reply:
> Also bumping to atleast v1.19.30 should be fine.. any reason why not?

I'm lazy?  Minimal viable implementation?  Something in that space ;).